src/boot/efi/secure-boot.c

```diff
@@ -126,3 +126,71 @@ EFI_STATUS secure_boot_enroll_at(EFI_FILE *root_dir, const char16_t *path) {
 
         return err;
 }
+
+static EFI_STATUS install_security_override_one(EFI_GUID *guid, SecurityOverride *override) {
+        EFI_STATUS err;
+
+        assert(guid);
+        assert(override);
+
+        _cleanup_free_ EFI_HANDLE *handles = NULL;
+        size_t n_handles = 0;
+
+        err = BS->LocateHandleBuffer(ByProtocol, guid, NULL, &n_handles, &handles);
+        if (err != EFI_SUCCESS)
+                /* No security arch protocol around? */
+                return err;
+
+        /* There should only ever be one security arch protocol instance, but let's be paranoid here. */
+        assert(n_handles == 1);
+
+        void *security = NULL;
+        err = BS->LocateProtocol(guid, NULL, &security);
+        if (err != EFI_SUCCESS)
+                return log_error_status_stall(err, u"Error getting security arch protocol: %r", err);
```

Comment:
Can we format GUIDs in sd-boot? If so it'd be nice to include the guid in the error message here

Reply:
Yes, but that'll go away once printf lands. And this error should never even happen as we've just been handed a list of handles that support this protocol anyways.